src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1018,7 +1018,12 @@ private bool OpenFile(WorkspaceInfo workspaceInfo, XmlDocument xmlDoc, out Works
             RegisterHomeWorkspace(newWorkspace);
            
             workspace = newWorkspace;
-
+            //this sets the event on Annotation. This event return the model from the workspace.
+            //When a model is ungrouped from a group, that model will be deleted from that group.
+            //So, when UNDO execution, cannot get that model from that group, it has to get from the workspace.
+            //The below method will set the event on every annotation model, that will return the specific model
+            //from workspace.
+            workspace.SetModelEventOnAnnotation();
```

Comment:
So, can you not undo `AnnotationModel` operations in a `CustomNodeWorkspaceModel`?